lib/engine/ability/reservation.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class Reservation < Base
+      attr_accessor :city, :tile
+      attr_reader :hex, :slot
+
+      def setup(hex:, city: 0, slot: 0)
+        @hex = hex
+        @city = city
+        @slot = slot
+        @tile = nil
+      end
+
+      def teardown
+        tile.cities[@city].reservations.delete(owner) if tile
```

Comment:
so reservations is an entity?

Reply:
Yeah. Previously it was just the name. With the entity it's easier to tell if the reservation belongs to a company that is owned by the corporation (e.g., for `reserved_by?`)